### ISSUE TITLE
New Rule: no-invalid-import-position

### ISSUE THREAD

**Issue by thecalamiity** [eslint/css#151]
### Rule details

Disallow invalid placement of @import rules.

### What type of rule is this?

Warns about a potential problem

### Example code

```css
a {}
@import 'foo.css';
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

`@import` rules must precede all other valid at-rules and style rules in a stylesheet (ignoring `@charset` and `@layer`), or else the `@import` rule is invalid.

Prior Art:
https://stylelint.io/user-guide/rules/no-invalid-position-at-import-rule/

**Comment by nzakas** [comment]
Sounds like a good idea to me, so marking as accepted.

I was surprised that CSSTree didn't pick this up in some way.
